pkg/webhook/workloadspread/validating/workloadspread_validation_test.go

```diff
@@ -943,3 +949,129 @@ func TestValidateWorkloadSpreadConflict(t *testing.T) {
 		})
 	}
 }
+
+func Test_validateWorkloadSpreadSubsets(t *testing.T) {
+	clonesetYaml := `
+apiVersion: apps.kruise.io/v1alpha1
+kind: CloneSet
+metadata:
+  name: test-cs
+spec:
+  replicas: 6
+  scaleStrategy:
+    disablePVCReuse: true
+  selector:
+    matchLabels:
+      app: test
+  template:
+    metadata:
+      labels:
+        app: test
+    spec:
+      containers:
+      - image: img:latest
+        imagePullPolicy: IfNotPresent
+        name: main
+        resources:
+          limits:
+            cpu: 10m
+            memory: 5Mi
+        volumeMounts:
+        - mountPath: /home/admin/logs
+          name: vol-1--0
+          subPath: logs
+      dnsPolicy: Default
+      restartPolicy: Always
+      terminationGracePeriodSeconds: 0
+  volumeClaimTemplates:
+  - metadata:
+      name: vol-1--0
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1Mi
+      storageClassName: alicloud-disk-topology-alltype
+    status:
+      phase: Pending
+`
+	stsYaml := `
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: test-sts
+spec:
+  serviceName: "nginx"
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: nginx
+        imagePullPolicy: IfNotPresent
+        name: main
+        resources:
+          limits:
+            cpu: 10m
+            memory: 5Mi
+        volumeMounts:
+        - mountPath: /home/admin/logs
+          name: vol-1--0
+          subPath: logs
+      dnsPolicy: Default
+      restartPolicy: Always
+      terminationGracePeriodSeconds: 0
+  volumeClaimTemplates:
+  - metadata:
+      annotations:
+        kubeone.ali/enable-cascading-deletion: "true"
+      name: vol-1--0
+    spec:
+      accessModes:
+      - ReadWriteOnce
+      resources:
+        requests:
+          storage: 1Mi
+    status:
+      phase: Pending
+`
+	cloneset := &appsv1alpha1.CloneSet{}
+	_ = yaml.Unmarshal([]byte(clonesetYaml), cloneset)
+	sts := &appsv1.StatefulSet{}
+	_ = yaml.Unmarshal([]byte(stsYaml), sts)
+	patchData := map[string]any{
+		"metadata": map[string]any{
+			"annotations": map[string]any{
+				"some-key": "some-value",
+			},
+		},
+	}
+	patch, _ := json.Marshal(patchData)
+	ws := &appsv1alpha1.WorkloadSpread{
+		Spec: appsv1alpha1.WorkloadSpreadSpec{
+			Subsets: []appsv1alpha1.WorkloadSpreadSubset{
+				{
+					Name: "test",
+					Patch: runtime.RawExtension{
+						Raw: patch,
+					},
+				},
+			},
+		},
+	}
+	errList := field.ErrorList{}
```

Comment:
plz add cases that we expect validateWorkloadSpreadSubsets will fail, and organize all cases using sub cases, you can refer to TestValidateWorkloadSpreadCreate